### PR TITLE
Remove ieTests

### PR DESCRIPTION
**Pull request by olpipi**
### Details:
 - Move cpp_interfaces tests from ieUnitTests to ov_unit_tests
 - Remove ieUnitTests and ieFuncTests

### Tickets:
 - CVS-73612

### CODE REVIEW

**Comment by olpipi** [comment]
ieFunc & Unit Tests are still enabled in Jenkins CI. tra

**Review by rkazants — DISMISSED** [review]
Why are tests removed?

File: .ci/azure/linux.yml [hunk 1]
```diff
@@ -437,13 +433,6 @@ jobs:
     displayName: 'MULTI UT'
     continueOnError: false
 
-  - script: |
-      # because of IR reader v7
-      export LD_LIBRARY_PATH=$(INSTALL_TEST_DIR)
-      $(RUN_PREFIX) $(INSTALL_TEST_DIR)/ieFuncTests --gtest_output=xml:$(INSTALL_TEST_DIR)/TEST-ieFuncTests.xml
-    displayName: 'IE FuncTests'
-    continueOnError: false
-
```
[the remark this answers is not in the document]
Comment (olpipi) [reply .ci/azure/linux.yml:446]: There is no tests in ieFuncTests anymore. They are all moved to other executables

**Comment by ilyachur** [comment]
> Why are tests removed?

Because these tests were moved to another target

**Review by ilyachur — COMMENTED** [review]
Please fix code style.
```
CMake Error at /home/runner/work/openvino/openvino/cmake/developer_package/clang_format/clang_format_check.cmake:15 (message):
  [clang-format] Code style check failed for:
  /home/runner/work/openvino/openvino/src/core/tests/cpp_interfaces/exception_test.cpp
```

Moreover why tests were moved to src/core/tests? It is a part of inference API. I think it should be moved to src/inference/tests/unit

File: .ci/azure/linux.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (ilyachur) [reply .ci/azure/linux.yml:446]: We have a new target for these tests

**Comment by rkazants** [comment]
@olpipi, please double-check that none test is lost.

**Review by ilyachur — APPROVED** [review]

**Pull request merged**